### ISSUE TITLE
Remove title bar and reposition actions.

### ISSUE THREAD

**Issue by dmarcos** [aframevr/aframe-inspector#214]
This came out after a conversation with @cvan @caseyyee and @ngokevin. We could get rid of the top menu bar to save space and move the actions closer to the context where they operate. For instance the export button could be somewhere in the scenegraph. We could add small toolbar in the place of the `scenegraph` title and using the space freed by the title bar. To add an entity there could just a big fat `+` icon that when clicked you'll see a list of options: Empty entity, shapes, primitives... Similar to the keynote UI that @cvan pointed out. 

![Keynote](https://media-mediatemple.netdna-ssl.com/wp-content/uploads/2015/07/01-document-size-opt.png)

**Comment by fernandojsg** [comment]
Ok, Keynote UI looks nice and clean indeed. My concerns is that they still have a menu on top of that for many other actions, they just don't have an icon for everything. For example even simple new/open/save you go to the menu. If we get rid of the menu we'll need to find place in the UI for many icons, and if we favour icons+labels or just labels maybe we won't end up saving so much space as we wanted.
I'll try to do some mockup with the current actions as we don't have many so far.

**Comment by dmarcos** [comment]
Maybe we will need to put them back but let's try living without them. I feel that those menus on top in most apps are just a hodgepodge consequence of lazy thinking.  The only advantage I see is that people have a place to go to look for an action when they're lost even if the clasification in different options rarely helps. I find myself in sublime chasing for actions I use pretty often: Was it in View, Files, Tools...?

**Comment by Coder206** [comment]
@dmarcos Agreed, however we should still consider some users may instinctively look for the menu option.

**Comment by fernandojsg** [comment]
@dmarcos yep I agree, but for example sublime text probably it's not a good example as you're a "power" user and you already use tons of shortcuts, this is probably not the kind of lower entry that we'll have in the editor. 
I also agree that menu are quite oldschool but you have a common and organised place to look for all the options you can do in your app.
Also we're assuming that we'll have the scenegraph always visible, that in many 3d programs I don't use to have as long as I don't need to check hierarchies, typically I hide it and I just work clicking on the objects around.
But as I said I'll do a mockup without the menu as we don't have as much options to take that decision yet.

**Comment by dmarcos** [comment]
True but if you don't break a little users old expectations it's hard to explore and try to find better ways. We are stuck in the past. Anyways, getting read of those menus is not that radical. In mobile (phone and tablets) you usually don't have them. Now that we don't have many actions we can go and find a better place for each of them. If we feel the need of those menus it could be an indication of feature creep. Often I think we should constrain the editor to DOM inspection / tweaking and maybe create separate tools for other purposes like an in VR editor.

**Comment by fernandojsg** [comment]
Yep I agree, also regarding the In VR editor as a separate tool (that of course you could inspect with the "2d" editor too :wink:)

**Comment by fernandojsg** [comment]
Btw right now we've the filter box in the scenegraph, what do you think about making it smaller and add the options to the right of the scenegraph? Or create another toolbar below or over the filter?

**Comment by dmarcos** [comment]
I personally like big fat search boxes. Another toolbar sounds better to me.

**Comment by fernandojsg** [comment]
Quick draft
<img width="1436" alt="screenshot 2016-07-17 01 04 22" src="https://cloud.githubusercontent.com/assets/782511/16897739/83a9e6a8-4bba-11e6-9ddd-7ed6f2b98fe2.png">

**Comment by dmarcos** [comment]
We can maybe get rid of the scenegraph/components titiles and rename `common` to `a-entity` or whatever tag. That way when you click on the scenegraph you will see your selection matched on the right. We have to also differentiate visually the common panel from the list of components

**Comment by dmarcos** [comment]
Instead of the floppy disk we could use a share icon 
![share icon](https://cdn4.iconfinder.com/data/icons/ionicons/512/icon-share-128.png). 
It can give you options to save to the clipboard, create a code pen or other hosting services.

**Comment by Coder206** [comment]
Those are some good ideas! I was wondering what was there reasoning behind the naming of the "Exit" button in the upper right hand corner (in white with red background)

**Comment by dmarcos** [comment]
The exit button is located in the same position than the edit button when the scene is running/playing. It allows you to go back and forth between edit and view modes. In edit mode al the dynamic behavior is paused. `Play Scene` might be better than `Exit`

**Comment by dmarcos** [comment]
We can use the icons for the components used in the scene-graph in the component titles as well. To reinforce the link between them

**Comment by Coder206** [comment]
@dmarcos Yes "Play scene" is definitely better here.

**Comment by fernandojsg** [comment]
I'm thinking about adding a toolbar to the components sidebar similar to the one that I've added on the scenegraph.
I would prefer to leave the `common` title as is, because that panel is about common components for the entity but it's not the "the entity itself" the whole entity is composed by the common and its custom components, so better to have the `a-entity` heading on top. Also having the actions outside the `common` panel makes more sense as for example `copy to clipboard` can be confusion if placed on `common` as people could think that you're going to copy just the `common`' components values.
Probably I'll take `ID` also from `common` and let the user edit directly on the title.
<img width="1436" alt="screenshot 2016-07-18 00 15 27" src="https://cloud.githubusercontent.com/assets/782511/16903401/5f64a490-4c7d-11e6-8a0a-9f0bca848dc1.png">

**Comment by dmarcos** [comment]
The common word does not give much info. An entity is rotation, position, scale, mixins, visible and a list of components. If you move the tag name to the top it conveys that everything underneath makes part of it I would just visually diferentiate the commons section from the list of components so they are not mixed.

**Comment by dmarcos** [comment]
Where are we relocating the exit button?

**Comment by fernandojsg** [comment]
Yep we need to make a different style for common, add component, and components themselves. We've some designs from Dennis about that. I'll add them another issue.
Regarding the exit button, I'm not sure where to locate it, at first I thought the best place was a corner but now we don't have free space, at least on the top, in the botton we could have if we add some kind of bar to one of the sidebar. Otherwise we could just add it centered on the botton?

**Comment by Coder206** [comment]
@fernandojsg Maybe there could be a side bar tabbed with the scenegraph for running the apps with a play and stop scene?

**Comment by fernandojsg** [comment]
@Coder206 can you make a mockup or draft in paper? I don't really understand what you mean with `for running the apps` and where to place it

**Comment by Coder206** [comment]
And performance information in this tabbed area

**Comment by fernandojsg** [comment]
I believe `play scene/exit` button should be always accesible and visible and it should match the same position as the `edit scene` when the scene is playing.

**Comment by Coder206** [comment]
@fernandojsg Ok sure! I agree, the button is an essential part of the UX and should be always accessible. Are your images above mockups in code?

**Comment by Coder206** [comment]
I did some drawings and without finishing them, I realized that my idea would simply be impractical. I was thinking of having the preview button in a hidden window can be accessible by a tab much like a browser tab would allow. I then thought how annoying this would be to access and not to mention you want to keep the button for starting and stopping the preview in a consistent position, so I think for the user, the best place for this toggle between inspector and preview should be bottom centered just as @fernandojsg mentioned.
